File: format_html.py
import re

with open("index.html", "r", encoding="utf-8") as f:
    content = f.read()

# Try to insert newlines to make it somewhat readable
formatted = re.sub(r'(<div|<section|<header|<nav|<footer|<main|<h1|<h2|<h3|<p|<a|<ul|<li|<button)', r'\n\1', content)
formatted = re.sub(r'(</div|</section|</header|</nav|</footer|</main|</h1|</h2|</h3|</p|</a|</ul|</li|</button>)', r'\1\n', formatted)
# Remove multiple newlines
formatted = re.sub(r'\n+', '\n', formatted)

with open("index.html", "w", encoding="utf-8") as f:
    f.write(formatted)

File: format_html.js
const fs = require('fs');
let content = fs.readFileSync('index.html', 'utf8');

content = content.replace(/(<div|<section|<header|<nav|<footer|<main|<h1|<h2|<h3|<p|<a|<ul|<li|<button)/g, '\n$1');
content = content.replace(/(<\/div|<\/section|<\/header|<\/nav|<\/footer|<\/main|<\/h1|<\/h2|<\/h3|<\/p|<\/a|<\/ul|<\/li|<\/button>)/g, '$1\n');
content = content.replace(/\n+/g, '\n');

fs.writeFileSync('index.html', content, 'utf8');
